### PR TITLE
Add visionOS support to MSAL iOS test targets and additional configs

### PR DESCRIPTION
**Pull request by mipetriu**
## Proposed changes

This PR adds support for MSAL iOS unit testing on visionOS.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by tkukielk** [comment]
am I allowed to review/comment?

File: MSAL/xcconfig/msal__static__lib__ios.xcconfig [hunk 1]
```diff
@@ -6,6 +6,7 @@ GENERATE_MASTER_OBJECT_FILE = YES
 // Add armv7s and arm64e to standard ARCHs.
 ARCHS = $(ARCHS_STANDARD) armv7s arm64e
 ARCHS[sdk=iphoneos*] = $(ARCHS_STANDARD) armv7s arm64e
+ARCHS[sdk=xros*] = $(ARCHS_STANDARD) armv7s arm64e
```
[the remark this answers is not in the document]
Comment (tkukielk) [reply MSAL/xcconfig/msal__static__lib__ios.xcconfig:9]: Somehow this one was posted?

**Comment by Ben-Pattinson** [comment]
So is this going to get merged in then? Can we have a new packager with it in please? It's a blocker for any service integration atm. I'm having to contemplate implementing it from scratch just to make headway.

File: MSAL/xcconfig/msal__static__lib__ios.xcconfig [hunk 2]
```diff
@@ -3,9 +3,7 @@
 // Force the linker to resolve symbols.
 GENERATE_MASTER_OBJECT_FILE = YES
 
-// Add armv7s and arm64e to standard ARCHs.
-ARCHS = $(ARCHS_STANDARD) armv7s arm64e
```
Comment (Veena11) [inline MSAL/xcconfig/msal__static__lib__ios.xcconfig:7]: why are these removed ?

Comment (tkukielk) [reply MSAL/xcconfig/msal__static__lib__ios.xcconfig:7]: armv7s is a 32-bit architecture no longer supported. arm64e is unused. $(ARCHS_STANDARD) is the default

**Review by Veena11 — APPROVED** [review]
